ArcGISRuntimeSDKQt_CppSamples/LocalServer/DynamicWorkspaceShapefile/DynamicWorkspaceShapefile.cpp

```diff
@@ -0,0 +1,153 @@
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#include "DynamicWorkspaceShapefile.h"
+
+#include "Map.h"
+#include "MapQuickView.h"
+#include "LocalServer.h"
+#include "DynamicWorkspace.h"
+#include "LocalMapService.h"
+#include "ShapefileWorkspace.h"
+#include "TableSublayerSource.h"
+#include "ArcGISMapImageSublayer.h"
+#include "ArcGISMapImageLayer.h"
+#include "SimpleRenderer.h"
+#include "SimpleFillSymbol.h"
+#include "SimpleLineSymbol.h"
+#include "SimpleMarkerSymbol.h"
+#include <QFile>
+#include <QFileInfo>
+#include <QDir>
+
+using namespace Esri::ArcGISRuntime;
+
+DynamicWorkspaceShapefile::DynamicWorkspaceShapefile(QQuickItem* parent /* = nullptr */):
+  QQuickItem(parent)
+{
+}
+
+void DynamicWorkspaceShapefile::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<DynamicWorkspaceShapefile>("Esri.Samples", 1, 0, "DynamicWorkspaceShapefileSample");
+}
+
+void DynamicWorkspaceShapefile::componentComplete()
+{
+  QQuickItem::componentComplete();
+
+  // find QML MapView component
+  m_mapView = findChild<MapQuickView*>("mapView");
+
+  // Create a map using the topographic basemap
+  m_map = new Map(Basemap::topographic(this), this);
+
+  // Set map to map view
+  m_mapView->setMap(m_map);
+
+  // Start the Local Server
+  if (LocalServer::instance()->isInstallValid())
+  {
+    if (LocalServer::instance()->status() == LocalServerStatus::Started)
+      emit localServerInitializationComplete(true);
+    else
+    {
+      connect(LocalServer::instance(), &LocalServer::statusChanged, this, [this]()
+      {
+        if (LocalServer::instance()->status() == LocalServerStatus::Failed)
+          emit localServerInitializationComplete(false);
+        else if (LocalServer::instance()->status() == LocalServerStatus::Started)
+          emit localServerInitializationComplete(true);
+      });
+      LocalServer::start();
+    }
+  }
+}
+
+// Start a service based on a file and folder path
+void DynamicWorkspaceShapefile::startLocalService(const QString& filePath, const QString& folder)
+{
+  if (!LocalServer::instance()->isInstallValid())
+    return;
+  // Setup file and folder variables
+  QFile f(filePath);
+  QFileInfo fileInfo(f.fileName());
+  QString workspacePath = QUrl(folder).toLocalFile();
+
+  // create a service from the blank MPK
+  QString mapPackagePath = QDir::homePath() + "/ArcGIS/Runtime/Data/mpk/mpk_blank.mpk";
+  if (m_localMapService)
+    m_localMapService->deleteLater();
+  m_localMapService = new LocalMapService(mapPackagePath, this);
```

Comment:
Please add empty line after the if case logic `m_localMapService->deleteLater();`